### PR TITLE
Lab12

### PR DESCRIPTION
**Pull request by bodnarira**

### CODE REVIEW

File: lab12.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import os
+import hashlib
+import threading
+import math
+
+
+thread_n = 20
+dir = "C:/Users/webvi/OneDrive/Documents/GitHub/"
```
Comment (polyakovartem) [inline lab12.py]: please do not use an absolute system-dependent path

File: lab12.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+duplicates = {}
+files_list = []
+for d in os.walk(dir):
+    for i in d[2]:
+        files_list.append(d[0]+"/"+i)
+
+files_shift = math.ceil(len(files_list)/thread_n)
+files_start = -files_shift
+threads = []
```
Comment (polyakovartem) [inline lab12.py]: this should be in a function

File: lab12.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+def get_duplicates (files_start, files_end, chunk_size):
+    for i in range(files_start, files_end):
+        hasher = hashlib.md5()
+        with open(files_list[i], 'rb') as file:
+            buf = file.read(chunk_size)
+        while len(buf) > 0:
+            hasher.update(buf)
+            buf = file.read(chunk_size)
+        file.close()
```
Comment (polyakovartem) [inline lab12.py]: why do you need this close?

**Review by polyakovartem — CHANGES_REQUESTED** [review]
and by the way, this a copy of your colleague's the work

**Comment by bodnarira** [comment]
I just asked for help to my friend in university and we worked with his computer. I can explain the algorithm. I know each thing what is written here.

File: lab12.py (hunk 3 above)
Comment (bodnarira) [reply lab12.py]: i don't need because of "with"

**Pull request closed without merging**